src/main/java/frc/robot/subsystems/VisionControl.java

```diff
@@ -74,17 +245,46 @@ private void update() {
         bally = visionData.by;
         ballr = visionData.br;
         wait = visionData.waitForOtherRobot;
+        chassis.updateEncoders();
     }
 
     private void calculateShooter() {
         // desiredAngle = __calculated_angle__;
         // desiredPower = __calculated_RPMS__;
+    }
+
+    private void calculateHoopChassis() {
+        hoop_rotation = 0.5 * (hoopr / 34.0);
+        //hoop_rotation = -pid.calculate(balla, 0);
+        if(hoopx < hoopRange[0]) {
```

Comment:
once hoop forward driving goes away, this method could be simplfied and renamed to something like `calculateHoopRotation`. Then it could return the calculated value instead of setting fields on the object, which is usually easier to think about and debug.

Reply:
done